piet-coregraphics/src/text.rs

```diff
@@ -416,18 +416,23 @@ impl TextLayout for CoreGraphicsTextLayout {
                 _ => false,
             })
             .count();
-        let height = typo_bounds.ascent + typo_bounds.descent + typo_bounds.leading;
         // this may not be exactly right, but i'm also not sure we ever use this?
         //  see https://stackoverflow.com/questions/5511830/how-does-line-spacing-work-in-core-text-and-why-is-it-different-from-nslayoutm
-        let cumulative_height =
-            (self.line_y_positions[line_number] + typo_bounds.descent + typo_bounds.leading).ceil();
+        let ascent = (typo_bounds.ascent + 0.5).floor();
+        let descent = (typo_bounds.descent + 0.5).floor();
+        let leading = (typo_bounds.leading + 0.5).floor();
+        let height = ascent + descent + leading;
+        let y_offset = self.line_y_positions[line_number] - ascent;
+        let cumulative_height = self.line_y_positions[line_number] + ascent + leading;
```

Comment:
Did you mean descent here? Actually I believe the invariant is that y_offset + height = cumulative_height, but I'm not 100% sure of this.

Reply:
yes, good catch.